### PR TITLE
Recommended career branch is found from quiz choices and outputted on new html page

### PR DESCRIPTION
**Pull request by lzoebaker**
Only a few questions have been added and there is no front-end design, but test out the functionality here: http://cs-career-step-2020.appspot.com/careerquiz.html
Selecting choices and clicking submit should lead to another html page that prints out the recommended career choice as a string.

### CODE REVIEW

File: src/main/java/com/google/sps/servlets/CareerQuizServlet.java [hunk 1]
```diff
@@ -17,21 +19,44 @@
...
+
+  @Override
+  /** when submit button on career quiz is clicked, this method is called and returns recommended path
+  based on selected choices*/
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    this.questionsAndChoices = this.careerQuestionDatabase.getQuestionsAndChoices();
+    String recommendedPath = handleQuizSubmission(request);
+    response.setContentType(JSON_CONTENT_TYPE);
+    String recommendedPathJson = gson.toJson(recommendedPath);
+    response.getWriter().println(recommendedPathJson);
```
Comment (DirtGrubDylan) [inline src/main/java/com/google/sps/servlets/CareerQuizServlet.java:49]: You can inline this `response.getWriter().println(gson.toJson(recommendedPath));`

File: src/main/webapp/choosebranch.html [hunk 2]
```diff
@@ -0,0 +1,10 @@
+<!doctype html>
+
+<html lang="en">
+<head>
+  <link rel="stylesheet" href="style.css">
+  <script src="choosebranch.js"></script>
+</head>
+
+<body onload="getRecommendedCareerPath()">
+</body>
```
Comment (DirtGrubDylan) [inline src/main/webapp/choosebranch.html:10]: New line

File: src/main/webapp/choosebranch.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+function handleResponse(response) {
+    // receives Java Object (ArrayList reponse)
+    const recommendedCareerPath = response.json();
+    recommendedCareerPath.then(changeDom);
+}
+
+function changeDom(recommendedCareerPath) {
+  console.log(recommendedCareerPath);
+}
```
Comment (DirtGrubDylan) [inline src/main/webapp/choosebranch.js:17]: New line. Does the linter catch this?

File: src/test/java/com/google/sps/CareerQuizTest.java [hunk 4]
```diff
@@ -40,49 +36,69 @@
...
- 
+
+  private CareerQuizServlet createCareerQuestionServlet() {
+    DatastoreService localDatastore = DatastoreServiceFactory.getDatastoreService();
+    careerQuestionDatabase = new CareerQuestionDatabase(localDatastore);
+    careerQuestionDatabase.putCareerQuestionAndChoicesIntoDatabase(
+        new CareerQuestionAndChoices(QUESTION, HARD_CODED_CHOICES));
+    CareerQuizServlet careerQuizServlet = new CareerQuizServlet(careerQuestionDatabase);
+    return careerQuizServlet;
+  }
```
Comment (DirtGrubDylan) [inline src/test/java/com/google/sps/CareerQuizTest.java:73]: white space between tests

File: src/main/java/com/google/sps/servlets/CareerQuizServlet.java (hunk 1 above)
Comment (lzoebaker) [reply src/main/java/com/google/sps/servlets/CareerQuizServlet.java:49]: done.

File: src/main/webapp/choosebranch.html (hunk 2 above)
Comment (lzoebaker) [reply src/main/webapp/choosebranch.html:10]: done.

File: src/main/webapp/choosebranch.js (hunk 3 above)
Comment (lzoebaker) [reply src/main/webapp/choosebranch.js:17]: honestly, need to investigate if the linter actually works at all..

Comment (lzoebaker) [reply src/main/webapp/choosebranch.js:17]: Changed.

File: src/test/java/com/google/sps/CareerQuizTest.java (hunk 4 above)
Comment (lzoebaker) [reply src/test/java/com/google/sps/CareerQuizTest.java:73]: done.

File: src/main/java/com/google/sps/servlets/CareerQuizServlet.java [hunk 5]
```diff
@@ -17,21 +19,43 @@
...
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    ArrayList<CareerQuestionAndChoices> questionsAndChoices = 
-        this.careerQuestionDatabase.getQuestionsAndChoices();
+    this.questionsAndChoices = this.careerQuestionDatabase.getQuestionsAndChoices();
     // Convert the ArrayLists to JSON
     Gson gson = new Gson();
     String questionsAndChoicesJson = gson.toJson(questionsAndChoices);
-    // Send the JSON as the response
-    response.setContentType(JSON_CONTENT_TYPE);
+    // Send the career questions and choices JSON as the response    response.setContentType(JSON_CONTENT_TYPE);
```
Comment (WillHipschman) [inline src/main/java/com/google/sps/servlets/CareerQuizServlet.java:37]: Remove if unneeded

File: src/main/java/com/google/sps/servlets/CareerQuizServlet.java [hunk 6]
```diff
@@ -17,21 +19,43 @@
...
     response.getWriter().println(questionsAndChoicesJson);
   }
+
+  @Override
+  /** when submit button on career quiz is clicked, this method is called and returns recommended path
```
Comment (WillHipschman) [inline src/main/java/com/google/sps/servlets/CareerQuizServlet.java:42]: Put javadoc above the @Override annotation

Comment (lzoebaker) [reply src/main/java/com/google/sps/servlets/CareerQuizServlet.java:42]: done.

**Pull request merged**